packages/agreement-process/test/agreementService.integration.test.ts

```diff
@@ -1434,4 +1473,131 @@ describe("Agreement service", () => {
       });
     });
   });
+  describe("archive agreement", () => {
+    it("should succeed when the requester is the consumer and the agreement is in a archivable state", async () => {
+      const authData = getRandomAuthData();
+      const eserviceId = generateId<EServiceId>();
+      const descriptorId = generateId<DescriptorId>();
+      const attributeId = generateId<AttributeId>();
+
+      const descriptor = buildDescriptorPublished(descriptorId, [
+        [buildEServiceAttribute(attributeId)],
+      ]);
+      const eservice = buildEService(eserviceId, authData.organizationId, [
+        descriptor,
+      ]);
+      const tenant = buildTenant(authData.organizationId);
+      const agreement = buildAgreement(
+        eserviceId,
+        tenant.id,
+        agreementState.active
+      );
+
+      await addOneEService(eservice, eservices);
+      await addOneTenant(tenant, tenants);
```

Comment:
Given that the archive does not test that eservice and tenant exist, you could also just add the agreement.

Doing something like the following should be sufficient to setup data for the success case without the need to create eservice or tenant ids manually or add them to the readmodel.

```typescript
   const agreement = {
      ...buildAgreement(),
      state: randomArrayItem(agreementArchivableStates),
    };
    await addOneAgreement(agreement, postgresDB, agreements);
    const authData = getRandomAuthData(agreement.consumerId);
```